### PR TITLE
Fixes map tile image cache issue

### PR DESCRIPTION
**Pull request by sam33rdhakal**
* Fixes the tile image cache issue (AIC-610)
* Removes the artic objects before updating so that removed objects in server are also removed by app (AIC-611)

### CODE REVIEW

File: image/build.gradle [hunk 1]
```diff
@@ -4,4 +4,6 @@ dependencies {
     kapt libs.glide_compiler
 
     api libs.rx_helpers
+    implementation ('com.github.bumptech.glide:okhttp3-integration:4.4.0@aar')
```
Comment (Cliabhach) [inline image/build.gradle]: Why aren't you using the same `glide_version` defined in the top-level build.gradle?

File: image/src/main/java/edu/artic/image/GlideModule.kt [hunk 2]
```diff
@@ -44,6 +54,25 @@ class GlideModule : AppGlideModule() {
...
+         * Configuration to use OkHttp client as the network client for Glide.
+         * OkHttp is also configured here to cache the image.
+         * Glide doesn't show the latest image if its from the same URL.
+         * OkHttp solves this problem by caching the image considering E-Tag and Last-Modified headers.
+         *
+         * @see https://github.com/bumptech/glide/issues/1847
+         */
+        val cacheSize: Long = 10 * 1024 * 1024 // 10 MiB
+        val clientBuilder = OkHttpClient.Builder()
+                .addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.HEADERS))
```
Comment (Cliabhach) [inline image/src/main/java/edu/artic/image/GlideModule.kt]: Please do not enable logging in release builds.

File: image/build.gradle [hunk 3]
```diff
@@ -4,4 +4,6 @@ dependencies {
...
+    implementation 'com.squareup.okhttp3:logging-interceptor:3.6.0'
```
Comment (Cliabhach) [inline image/build.gradle]: Please use `debugImplementation` or similar.

File: db/src/main/kotlin/edu/artic/db/AppDataManager.kt [hunk 4]
```diff
@@ -169,6 +169,7 @@ class AppDataManager @Inject constructor(
                                         }
                                     }
                                 }
+                                objectDao.clear()
```
Comment (Cliabhach) [inline db/src/main/kotlin/edu/artic/db/AppDataManager.kt]: Based on this, do we need to clear out the gallery and audio file DAOs as well?

File: image/build.gradle (hunk 1 above)
Comment (sam33rdhakal) [reply image/build.gradle]: Updated.

File: image/build.gradle (hunk 3 above)
Comment (sam33rdhakal) [reply image/build.gradle]: Nice catch! Updated

File: image/src/main/java/edu/artic/image/GlideModule.kt (hunk 2 above)
Comment (sam33rdhakal) [reply image/src/main/java/edu/artic/image/GlideModule.kt]: Now interceptor will only be added if build is `Debug`.

File: image/build.gradle (hunk 3 above)
Comment (Cliabhach) [reply image/build.gradle]: huh. forgot we couldn't use that without splitting out code. guess it's fine as-is

**Pull request merged**